src/nile/core/account.py

```diff
@@ -79,29 +79,58 @@ def deploy(self):
 
     def send(self, address_or_alias, method, calldata, max_fee, nonce=None):
         """Execute a tx going through an Account contract."""
-        if not is_alias(address_or_alias):
-            address_or_alias = normalize_number(address_or_alias)
+        # get target address with the right format
+        target_address = self._get_target_address(address_or_alias)
 
-        target_address, _ = (
-            next(deployments.load(address_or_alias, self.network), None)
-            or address_or_alias
+        # process and parse arguments
+        calldata, max_fee, nonce = self._process_arguments(calldata, max_fee, nonce)
+
+        calldata, sig_r, sig_s = self.signer.sign_transaction(
+            sender=self.address,
+            calls=[[target_address, method, calldata]],
+            nonce=nonce,
+            max_fee=max_fee,
+            version=TRANSACTION_VERSION,
         )
 
-        calldata = [int(x) for x in calldata]
+        return call_or_invoke(
+            contract=self,
+            type="invoke",
+            method="__execute__",
+            params=calldata,
+            network=self.network,
+            signature=[str(sig_r), str(sig_s)],
+            max_fee=str(max_fee),
+        )
 
-        if nonce is None:
-            nonce = get_nonce(self.address, self.network)
+    def simulate(self, address_or_alias, method, calldata, max_fee, nonce=None):
+        """Simulate a tx going through an Account contract."""
+        return self.execute_query(
+            "simulate", address_or_alias, method, calldata, max_fee, nonce
+        )
 
-        if max_fee is None:
-            max_fee = 0
-        else:
-            max_fee = int(max_fee)
+    def estimate_fee(self, address_or_alias, method, calldata, max_fee, nonce=None):
+        """Estimate fee for a tx going through an Account contract."""
+        return self.execute_query(
+            "estimate_fee", address_or_alias, method, calldata, max_fee, nonce
+        )
+
+    def execute_query(
```

Comment:
what do you think about renaming this `execute` or something similar, and to also accept a `transaction` type that actually executes it, to avoid code duplication with `send`